### PR TITLE
Docs: Remove env_developer.yml from install guide

### PR DESCRIPTION
**Pull request by tovogt**
Changes proposed in this PR:
- Following #712, the installation instructions for developers were outdated. This fixes the install guide.

### PR Author Checklist

- [x] Read the [Contribution Guide][contrib]
- [x] Correct target branch selected (if unsure, select `develop`)
- [x] Descriptive pull request title added
- [x] Source branch up-to-date with target branch
- [x] [Documentation](https://climada-python.readthedocs.io/en/latest/guide/Guide_PythonDos-n-Donts.html#2.--Commenting-&-Documenting) updated
- [ ] [Tests][testing] updated
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]
- [ ] [Changelog](https://github.com/CLIMADA-project/climada_python/blob/main/CHANGELOG.md) updated

### PR Reviewer Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] [CLIMADA Reviewer Checklist](https://climada-python.readthedocs.io/en/latest/guide/Guide_Reviewer_Checklist.html) passed
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

[contrib]: https://github.com/CLIMADA-project/climada_python/blob/main/CONTRIBUTING.md
[testing]: https://climada-python.readthedocs.io/en/latest/guide/Guide_Continuous_Integration_and_Testing.html
[linter]: https://climada-python.readthedocs.io/en/stable/guide/Guide_Continuous_Integration_and_Testing.html#3.C.--Static-Code-Analysis

### CODE REVIEW

**Comment by tovogt** [comment]
Ah, okay, I missed that part. I thought I was in the section about developer instructions already. And the dev requirements used to be the default here before. But I'm fine with removing the extra part.

File: doc/guide/install.rst [hunk 1]
```diff
@@ -118,26 +118,27 @@ For advanced Python users or developers of CLIMADA, we recommed cloning the CLIM
...
 
    .. code-block:: shell
 
       conda env create -n climada_env -f requirements/env_climada.yml
-      conda env update -n climada_env -f requirements/env_developer.yml
       conda activate climada_env
 
-#. Install the local CLIMADA source files as Python package using ``pip``:
+#. Install the local CLIMADA source files as Python package using ``pip``, including additional
+   requirements for developers:
```
Comment (tovogt) [inline doc/guide/install.rst:130]:
```suggestion
#. Install the local CLIMADA source files as Python package using ``pip``:
```

File: doc/guide/install.rst [hunk 2]
```diff
@@ -118,26 +118,27 @@ For advanced Python users or developers of CLIMADA, we recommed cloning the CLIM
...
 
    .. code-block:: shell
 
-      python -m pip install -e ./
+      python -m pip install -e .[dev]
```
Comment (tovogt) [inline doc/guide/install.rst:134]:
```suggestion
      python -m pip install -e ./
```

**Comment by tovogt** [comment]
So, I added you suggested changes.

Are you aware that the `Install Developer Dependencies (Optional)` claims that `[test]` is required to run the tests, but you already run tests as an example at the end of the previous `Advanced Instructions` where this extra is not installed yet.

**Comment by peanutfun** [comment]
@tovogt I admit, this is a bit imprecise. Executing `python -m unittest climada.engine.test.test_impact` does not require the test extras. These extras mostly contain "test tools" like `coverage` etc.

**Comment by peanutfun** [comment]
I tried to mention this somewhat explicitly:

> Building the documentation and running the **entire test suite** of CLIMADA requires additional dependencies

(emphasis by me)

@tovogt Do you think this is clear enough or should we change something about that?

**Comment by tovogt** [comment]
Don't worry, I think it's clear enough. Thanks for considering my concerns :)

**Comment by peanutfun** [comment]
Great, thanks a lot for taking your time and fixing the docs! 🎉

**Review by peanutfun — APPROVED** [review]

**Pull request merged**